File: app/src/main/java/com/rafaelmfer/nasaexperience/ui/activity/ActivityHome.java
package com.rafaelmfer.nasaexperience.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.FragmentManager;

import com.google.android.material.navigation.NavigationView;
import com.rafaelmfer.nasaexperience.R;

import org.jetbrains.annotations.NotNull;

public class ActivityHome extends AppCompatActivity implements  NavigationView.OnNavigationItemSelectedListener {

    ImageButton ibMenu;
    Button btMarsWeather, btAsteroids, btAstronomicImage;
    NavigationView navigationView;
    DrawerLayout drawerLayout;
    FragmentManager manager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        bindViews();

        ibMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                drawerLayout.openDrawer(GravityCompat.START);
            }
        });

        btMarsWeather.setOnClickListener(clickToStartNewActivity(ActivityMarsWeather.class));
        btAsteroids.setOnClickListener(clickToStartNewActivity(ActivityAsteroidsNearFromEarth.class));
        btAstronomicImage.setOnClickListener(clickToStartNewActivity(ActivityAstronomicImageOfTheDay.class));
    }

    private void bindViews() {
        ibMenu = findViewById(R.id.ibMenu);
        btMarsWeather = findViewById(R.id.btMarsWeather);
        btAsteroids = findViewById(R.id.btAsteroids);
        btAstronomicImage = findViewById(R.id.btAstronomicImage);
        ibMenu = findViewById(R.id.ibMenu);
        navigationView = findViewById(R.id.nav_view);
        drawerLayout = findViewById(R.id.drawer_layout);
        navigationView.setNavigationItemSelectedListener(this);
        manager = getSupportFragmentManager();
    }

    @NotNull
    private View.OnClickListener clickToStartNewActivity(Class<?> activity) {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(view.getContext(), activity));
            }
        };
    }
    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.nav_favorites:
                Toast.makeText(this,"Me faz a " + "Lei Audio" + "!",Toast.LENGTH_LONG).show();
                break;
            case R.id.nav_log_out:
                Toast.makeText(this,"Me faz um LÊ ALTO!",Toast.LENGTH_LONG).show();
                break;
            case R.id.nav_rate_avaliation:
                Toast.makeText(this,"Me faz um Le'ato!",Toast.LENGTH_LONG).show();
                break;
        }
        drawerLayout.closeDrawer(GravityCompat.START);
        return false;
    }

}

File: app/src/main/java/com/rafaelmfer/nasaexperience/ui/activity/ActivityHome.kt
package com.rafaelmfer.nasaexperience.ui.activity

import android.app.AlertDialog
import android.content.Intent
import android.view.MenuItem
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.activity.viewModels
import androidx.core.view.GravityCompat
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInOptions
import com.google.android.material.navigation.NavigationView
import com.rafaelmfer.nasaexperience.R
import com.rafaelmfer.nasaexperience.baseviews.ActBind
import com.rafaelmfer.nasaexperience.databinding.ActivityHomeBinding
import com.rafaelmfer.nasaexperience.debugging.ExceptionHandler
import com.rafaelmfer.nasaexperience.extensions.CircleTransform
import com.rafaelmfer.nasaexperience.extensions.addMarginTopStatusBarHeight
import com.rafaelmfer.nasaexperience.extensions.setFullScreen
import com.rafaelmfer.nasaexperience.extensions.toast
import com.rafaelmfer.nasaexperience.viewmodel.ViewModelLoginRegisterFirebase
import com.squareup.picasso.Picasso

class ActivityHome : ActBind<ActivityHomeBinding>(), NavigationView.OnNavigationItemSelectedListener {

    private val viewModelLoginFirebase: ViewModelLoginRegisterFirebase by viewModels()

    private val loginIntentGoogle by lazy {
        GoogleSignIn.getClient(
            this, GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                .requestIdToken(getString(R.string.default_web_client_id))
                .requestEmail()
                .build()
        )
    }

    override val bindClass: Class<ActivityHomeBinding> by lazy { ActivityHomeBinding::class.java }

    override fun ActivityHomeBinding.onBoundView() {
        exceptionHandler = ExceptionHandler::class.java
        setFullScreen()
        binding.contentHome.apply {
            menuNavdrawer.run {
                addMarginTopStatusBarHeight()
                setOnClickListener { drawerLayout.openDrawer(GravityCompat.START) }
            }
            btMarsWeather.setOnClickListener(clickToStartNewActivity(ActivityMarsWeather::class.java))
            btAsteroids.setOnClickListener(clickToStartNewActivity(ActivityAsteroidsNearFromEarth::class.java))
            btAstronomicImage.setOnClickListener(clickToStartNewActivity(ActivityAstronomicImageOfTheDay::class.java))
        }
        navigationView.setNavigationItemSelectedListener(this@ActivityHome)
        val hView = navigationView.getHeaderView(0)
        val profileName = hView.findViewById<View>(R.id.profile_name) as TextView
        val profileEmail = hView.findViewById<View>(R.id.profile_email) as TextView
        val profileImage = hView.findViewById<View>(R.id.profile_image) as ImageView
        val currentUser = viewModelLoginFirebase.firebaseUser
        profileName.text = currentUser?.displayName
        profileEmail.text = currentUser?.email
        Picasso.get().load(currentUser?.photoUrl)
            .resize(100, 100)
            .transform(CircleTransform())
            .centerCrop()
            .into(profileImage)
    }

    private fun clickToStartNewActivity(activity: Class<*>) =
        View.OnClickListener { startActivity(Intent(it.context, activity)) }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_favorites -> {
                openFavoritesScreen()
            }
            R.id.nav_log_out -> {
                logoffFire()
            }
            R.id.nav_about -> {
                openAboutScreen()
            }
        }
        binding.drawerLayout.closeDrawer(GravityCompat.START)
        return false
    }

    private fun openFavoritesScreen() {
        //TODO ABRIR BOTTOM SHEET FUTURAMENTE AQUI
        //TODO POR ENQUANTO ABRE ESSA ACTIVITY MESMO ( ͡° ʖ̯ ͡°)
        startActivity(Intent(this@ActivityHome, ActivityFavorites::class.java))
    }

    private fun openAboutScreen() {
        startActivity(Intent(this@ActivityHome, ActivityAbout::class.java))
    }

    private fun logoffFire() {
        viewModelLoginFirebase.logoffFirebase()
        loginIntentGoogle.revokeAccess()
        onBackPressed()
    }

    override fun onBackPressed() {

        val builder = AlertDialog.Builder(this)
        builder.setMessage("Deseja sair da nave?")
        builder.setCancelable(true)
        builder.setNegativeButton("Cancelar") { dialogInterface, cancel ->

            dialogInterface.cancel()
        }
        builder.setPositiveButton("Sair") { dialogInterface, exit ->
            toast(getString(R.string.dialog_exite_message))
            finishAffinity()
        }

        val alertDialog = builder.create()
        alertDialog.show()
    }
}